### PR TITLE
Consume elemental-cli in place of yip

### LINKED ISSUE #662: Enhance cloud-init execution output
**Is your feature request related to a problem? Please describe.**
When having errors in parsing cloud-init file, it's hard to tell which files have issues. For example, from the following output:

```
ERRO[0000] fatal error: 1 error occurred:
	* yaml: line 8: did not find expected key
``` 
It's hard to tell which is the file causing the error

**Describe the solution you'd like**
A more informative output, including the file which is failing to load

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #942: `elemental cloud-init`
- [x] reuse `yip` as a library
- [x] Cut a new elemental-cli version (https://github.com/rancher-sandbox/elemental-cli/releases/tag/v0.0.2)
- [ ] Integrate `elemental cloud-init` in place of `yip` inside `cOS-toolkit` #976

### PR DESCRIPTION
**Pull request by mudler**
This PR is multi-fold:
- It replaces yip with elemental-cli in the new package versions
- It moves elemental-cli under `toolchain`. Renames the output binary to
  /usr/bin/elemental for sake of semplicity.
- Updates test accordingly
- The new underlying yip version changes the output to be more focused
  on the executed stages. This makes obsolete #662.

Fixes #942
Closes #662
Related to https://github.com/rancher-sandbox/cOS-toolkit/issues/108

Signed-off-by: Ettore Di Giacinto <edigiacinto@suse.com>

### CODE REVIEW

File: packages/cos-setup/definition.yaml [hunk 1]
```diff
@@ -2,6 +2,6 @@ name: cos-setup
 category: system
 version: "0.5"
```
Comment (mudler) [inline packages/cos-setup/definition.yaml:3]:
```suggestion
version: "0.6"
```

File: packages/immutable-rootfs/30cos-immutable-rootfs/module-setup.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ install() {
     declare initdir="${initdir}"
```
Comment (mudler) [inline packages/immutable-rootfs/30cos-immutable-rootfs/module-setup.sh:24]: Package needs a new version, too

File: packages/toolchain/elemental-cli/definition.yaml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 name: "elemental-cli"
-category: "utils"
-version: 0.0.1
+category: "toolchain"
+bin_name: "elemental"
```
Comment (Itxaka) [inline packages/toolchain/elemental-cli/definition.yaml]: cant we just rename the binary in the elemental-cli side instead? So its simpler?

Comment (mudler) [reply packages/toolchain/elemental-cli/definition.yaml]: we shall rename the repository, no ?

Comment (mudler) [reply packages/toolchain/elemental-cli/definition.yaml]: nothing we should worry about for now, there is one elemental binary now.. a gui would probably go in another project, etc

Comment (mudler) [reply packages/toolchain/elemental-cli/definition.yaml]: Opened https://github.com/rancher-sandbox/cOS-toolkit/issues/978 to track that separately

**Review by davidcassany — APPROVED** [review]
Looks good to me

I am glad now we did not code a new console for yip in elemental, minimize the chances for regressions 😄

**Pull request merged**